crates/legacy/src/legacy_testing/mod.rs

```diff
@@ -0,0 +1,278 @@
+    type BlockPayload = TestBlockPayload;
+    type SignatureKey = BLSPubKey;
+    type Transaction = TestTransaction;
+    type ValidatedState = TestValidatedState;
+    type InstanceState = TestInstanceState;
+    type Membership = StaticCommittee<Self>;
+    type BuilderSignatureKey = BuilderKey;
+    type AuctionResult = TestAuctionResult;
+}
+
+pub async fn start_builder_state_without_event_loop(
+    channel_capacity: usize,
+    num_storage_nodes: usize,
+) -> (
+    BroadcastSender<MessageType<TestTypes>>,
+    Arc<RwLock<GlobalState<TestTypes>>>,
+    BuilderState<TestTypes>,
+) {
+    // set up the broadcast channels
+    let (bootstrap_sender, bootstrap_receiver) =
+        broadcast::<MessageType<TestTypes>>(channel_capacity);
+    let (_decide_sender, decide_receiver) = broadcast::<MessageType<TestTypes>>(channel_capacity);
+    let (_da_sender, da_receiver) = broadcast::<MessageType<TestTypes>>(channel_capacity);
+    let (_quorum_sender, quorum_proposal_receiver) =
+        broadcast::<MessageType<TestTypes>>(channel_capacity);
+    let (senders, _receivers) = broadcast::<MessageType<TestTypes>>(channel_capacity);
+    let (tx_sender, tx_receiver) =
+        broadcast::<Arc<ReceivedTransaction<TestTypes>>>(channel_capacity);
+
+    let genesis_vid_commitment = vid_commitment(&[], num_storage_nodes);
+    let genesis_builder_commitment = BuilderCommitment::from_bytes([]);
+
+    // instantiate the global state
+    let global_state = Arc::new(RwLock::new(GlobalState::<TestTypes>::new(
+        bootstrap_sender,
+        tx_sender.clone(),
+        genesis_vid_commitment,
+        ViewNumber::genesis(),
+        ViewNumber::genesis(),
+        10,
+    )));
+
+    // instantiate the bootstrap builder state
+    let builder_state = BuilderState::new(
+        ParentBlockReferences {
+            view_number: ViewNumber::new(0),
+            vid_commitment: genesis_vid_commitment,
+            leaf_commit: Commitment::<Leaf<TestTypes>>::default_commitment_no_preimage(),
+            builder_commitment: genesis_builder_commitment,
+        },
+        decide_receiver.clone(),
+        da_receiver.clone(),
+        quorum_proposal_receiver.clone(),
+        bootstrap_receiver,
+        tx_receiver,
+        VecDeque::new(),
+        global_state.clone(),
+        NonZeroUsize::new(num_storage_nodes).unwrap(),
+        Duration::from_millis(100),
+        1,
+        Arc::new(TestInstanceState::default()),
+        Duration::from_millis(100),
+        Arc::new(TestValidatedState::default()),
+    );
+
+    (senders, global_state, builder_state)
+}
+
+/// get transactions submitted in previous rounds, [] for genesis
+/// and simulate the block built from those
+pub async fn calc_proposal_msg(
+    num_storage_nodes: usize,
+    round: usize,
+    prev_quorum_proposal: Option<QuorumProposal<TestTypes>>,
+    transactions: Vec<TestTransaction>,
+) -> (
+    QuorumProposal<TestTypes>,
+    MessageType<TestTypes>,
```

Comment:
Nit: can we replace these two `MessageType`s with underlying concrete message types and wrap them at the call site? I think this would make function signature more clear at a glance

Reply:
Updated in [1d14f1f](https://github.com/EspressoSystems/marketplace-builder-core/pull/176/commits/1d14f1f81eacfc86ec8baf62394d5ca28161c825)